pkg/odo/cli/component/list.go

```diff
@@ -256,43 +270,60 @@ func (lo *ListOptions) Run() (err error) {
 				if err != nil {
 					return err
 				}
-				componentList = append(componentList, comps.Items...)
+				components = append(components, comps.Items...)
 			}
-			// Get machine readable component list format
-			components = component.GetMachineReadableFormatForList(componentList)
 		} else {
 
-			components, err = component.List(lo.Client, lo.Application, lo.LocalConfigInfo)
+			componentList, err := component.List(lo.Client, lo.Application, lo.LocalConfigInfo)
+			// compat
+			components = componentList.Items
 			if err != nil {
 				return errors.Wrapf(err, "failed to fetch component list")
 			}
 		}
+	}
 
-		if log.IsJSON() {
-			machineoutput.OutputSuccess(components)
-		} else {
-			if len(components.Items) != 0 {
-				if lo.hasDevfileComponents {
-					fmt.Println()
-				}
-				lo.hasS2IComponents = true
-				w := tabwriter.NewWriter(os.Stdout, 5, 2, 3, ' ', tabwriter.TabIndent)
-				fmt.Fprintln(w, "Openshift Components: ")
-				fmt.Fprintln(w, "APP", "\t", "NAME", "\t", "PROJECT", "\t", "TYPE", "\t", "SOURCETYPE", "\t", "STATE")
-				for _, comp := range components.Items {
-					fmt.Fprintln(w, comp.Spec.App, "\t", comp.Name, "\t", comp.Namespace, "\t", comp.Spec.Type, "\t", comp.Spec.SourceType, "\t", comp.Status.State)
-				}
-				w.Flush()
+	w := tabwriter.NewWriter(os.Stdout, 5, 2, 3, ' ', tabwriter.TabIndent)
+
+	if !log.IsJSON() {
+
+		if len(devfileComponents) != 0 {
+			lo.hasDevfileComponents = true
+			fmt.Fprintln(w, "Devfile Components: ")
+			fmt.Fprintln(w, "APP", "\t", "NAME", "\t", "PROJECT", "\t", "TYPE", "\t", "STATE")
+			for _, comp := range devfileComponents {
+				fmt.Fprintln(w, comp.Spec.App, "\t", comp.Spec.Name, "\t", comp.Namespace, "\t", comp.Spec.Type, "\t", comp.Status.State)
 			}
+			w.Flush()
+
+		}
+		if lo.hasDevfileComponents {
+			fmt.Fprintln(w)
+		}
+
+		if len(components) != 0 {
+			if lo.hasDevfileComponents {
+				fmt.Println()
+			}
+			lo.hasS2IComponents = true
+			w := tabwriter.NewWriter(os.Stdout, 5, 2, 3, ' ', tabwriter.TabIndent)
+			fmt.Fprintln(w, "Openshift Components: ")
```

Comment:
why we print `OpenShift Components` & `DevfileComponents`? devfile components can also be openshift components.